api/base/serializers.py

```diff
@@ -758,6 +758,27 @@ def is_valid(self, clean_html=True, **kwargs):
 
         return ret
 
+class JSONAPIRelationshipSerializer(ser.Serializer):
+
+    def to_representation(self, obj, envelope='data'):
+        ret = {}
+        meta = getattr(self, 'Meta', None)
+        type_ = getattr(meta, 'type_', None)
+        assert type_ is not None, 'Must define Meta.type_'
+
+        data = collections.OrderedDict([
+            ('links', collections.OrderedDict()),
+            ('data', collections.OrderedDict()),
+        ])
+        relation_id_field = self.fields['id']
+
+        relationship = relation_id_field.to_representation(relation_id_field.get_attribute(obj))
```

Comment:
I like this a lot because it's how the Django Rest Framework serializes the fields.

For readability, split it into two lines:
`attribute = relation_id_field.get_attribute(obj)`
`relationship = relation_id_field.to_representation(attribute)`